app/assets/javascripts/admin.js

```diff
@@ -24,7 +24,8 @@ var assignmentsReady = function(){
 
 }
 
-$(document).ready(assignmentsReady)
+$(document).ready(assignmentsReady);
+$(document).on('page:load', assignmentsReady);
```

Comment:
> on next firing

I'm sure what exactly you mean by this. Which event? To be perfectly clear:
```javascript
// turbolinks disabled
$(document).ready(func)
```
is equivalent to 
```javascript
// turbolinks enabled
$(document).ready(func); // only fires if turbolinks does not have this page cached
$(document).on('page:load', func); // fires if turbolinks has the page cached
```

This is the desired behavior in this case. @thecristen